### PR TITLE
move chat implementation to src/

### PR DESCRIPTION
**Pull request by u9g**

### CODE REVIEW

File: src/client/play.js [hunk 1]
```diff
@@ -5,12 +5,14 @@ const concat = require('../transforms/binaryStream').concat
...
 
+  const EMPTY_BUFFER = Buffer.from([])
+
   function onLogin (packet) {
     client.state = states.PLAY
     client.uuid = packet.uuid
     client.username = packet.username
     client.signMessage = (message, timestamp, salt = 0) => {
-      if (!client.profileKeys) throw Error("Can't sign message without profile keys, please set valid auth mode")
+      if (!client.profileKeys) return EMPTY_BUFFER
```
Comment (extremeheat) [inline src/client/play.js:15]: this should be an exception as no signature is returned

File: src/client/chat.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  function onChat (packet, isMessageFromServer = false) {
+    let message = packet.message ?? packet.unsignedChatContent ?? packet.signedChatContent
+    if (isMessageFromServer) message = packet.content
+
+    let verified = isMessageFromServer ? null : false // default to false if this is a system chat
+    if (!isMessageFromServer && client.profileKeys != null && packet.signature) {
+      verified = client.verifyMessage(players[packet.senderUuid].publicKey, packet)
+    }
+    client.emit('chat_received', { verified, message, isMessageFromServer })
```
Comment (extremeheat) [inline src/client/chat.js:15]: Why not a chatMessage and serverMessage event ? In mineflayer we don't handle both in one function and the event emitted has a type in it. If chat abstraction is moving to nmp just copy over the existing mineflayer implementation / events

File: src/client/play.js (hunk 1 above)
Comment (u9g) [reply src/client/play.js]: Fixed in latest commit

**Comment by rom1504** [comment]
done in another pr

**Pull request closed without merging**